jerry-core/vm/vm.c

```diff
@@ -1651,6 +1657,36 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_ADD:
         {
+          if (ecma_are_values_integer_numbers (left_value, right_value))
```

Comment:
Could work, but I need to make a separate patch to try that out and see what is happening.